TumbleBitSetup/PoupardStern.cs

```diff
@@ -0,0 +1,263 @@
+﻿using Org.BouncyCastle.Math;
+using Org.BouncyCastle.Security;
+using Org.BouncyCastle.Utilities;
+using System;
+
+
+namespace TumbleBitSetup
+{
+    public class PoupardStern
+    {
+        /// <summary>
+        /// Proving Algorithm specified in page 12 (3.2.1) of the setup
+        /// </summary> 
+        /// <param name="p">P in the secret key</param>
+        /// <param name="q">Q in the secret key</param>
+        /// <param name="e">Public Exponent in the public key</param>
+        /// <param name="ps">The public string from the setup</param>
+        /// <param name="k">Security parameter as specified in the setup.</param>
+        /// <returns>List of x values and y</returns>
+        public static Tuple<BigInteger[], BigInteger> Proving(BigInteger p, BigInteger q, BigInteger e, byte[] psBytes, int k = 128)
+        {
+            BigInteger y;
+            BigInteger Two = BigInteger.Two;
+
+            // Generate a keyPair from p, q and e
+            var keyPair = new RsaKey(p, q, e);
+            var pubKey = new RsaPubKey(keyPair._pubKey);
+            var secKey = keyPair._privKey;
+
+            BigInteger Modulus = pubKey._pubKey.Modulus;
+            int ModulusBitLength = Modulus.BitLength;
+
+            // Calculating 2^{ |N| - 1}
+            BigInteger lowerLimit = Two.Pow(ModulusBitLength - 1);
+            
+            // Calculating phi
+            BigInteger pSub1 = p.Subtract(BigInteger.One);
+            BigInteger qSub1 = q.Subtract(BigInteger.One);
+            BigInteger phi = pSub1.Multiply(qSub1);
+
+            // Check that N>2^{|N|-1}
```

Comment:
From what I found in BouncyCastle ([here](https://github.com/bcgit/bc-csharp/blob/master/crypto/src/math/BigInteger.cs#L1042)) is that
a.CompareTo(b) will return
1 if a > b
0 if a == b
-1 if a < b.